ami/install.bash

```diff
@@ -1,46 +1,52 @@
 #!/usr/bin/env bash
 set -euo pipefail
 
-# Enable Jessie backports to install the latest Java JRE.
-cat <<EOF >>/etc/apt/sources.list
-
-# Debian backports
-deb http://httpredir.debian.org/debian jessie-backports main
-EOF
+apt-get install -y dirmngr
 
 # Add sbt repo.
 echo "deb https://dl.bintray.com/sbt/debian /" | tee -a /etc/apt/sources.list.d/sbt.list
 apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv 2EE0EA64E40A89B84B2DF73499E82A75642AC823
 
 # Add Docker repo.
-echo "deb https://apt.dockerproject.org/repo debian-jessie main" | tee -a /etc/apt/sources.list.d/docker.list
+echo "deb https://apt.dockerproject.org/repo debian-stretch main" | tee -a /etc/apt/sources.list.d/docker.list
 apt-key adv --keyserver hkp://p80.pool.sks-keyservers.net:80 --recv-keys 58118E89F3A912897C070ADBF76221572C52609D
 
 # Add Mesos repo.
 apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv E56151BF && \
-  echo "deb http://repos.mesosphere.com/debian jessie-unstable main" | tee -a /etc/apt/sources.list.d/mesosphere.list && \
-  echo "deb http://repos.mesosphere.com/debian jessie-testing main" | tee -a /etc/apt/sources.list.d/mesosphere.list && \
-  echo "deb http://repos.mesosphere.com/debian jessie main" | tee -a /etc/apt/sources.list.d/mesosphere.list && \
-
-  apt-get -y update
+  echo "deb http://repos.mesosphere.com/debian stretch-unstable main" | tee -a /etc/apt/sources.list.d/mesosphere.list && \
+  echo "deb http://repos.mesosphere.com/debian stretch-testing main" | tee -a /etc/apt/sources.list.d/mesosphere.list && \
+  echo "deb http://repos.mesosphere.com/debian stretch main" | tee -a /etc/apt/sources.list.d/mesosphere.list
+apt-get -y update
 
 # Add github.com to known hosts
 ssh-keyscan github.com >> /home/admin/.ssh/known_hosts
 ssh-keyscan github.com >> /root/.ssh/known_hosts
 
 # Install dependencies
-apt install -t jessie-backports -y openjdk-8-jdk
-update-java-alternatives -s java-1.8.0-openjdk-amd64
-
 apt-get install -y \
         build-essential \
         curl \
         docker-engine \
         git \
-        npm \
-        python3-pip \
-        rpm \
-        sbt
+        openjdk-8-jdk \
+        libssl-dev \
+        sbt \
+        zlib1g-dev
+
+# Download, compile and install Python 3.6.2
```

Comment:
How come install python from sources and not using a binary release? (not that it's changing that much, just curious).

Reply:
There is no Python 3.6 backport for Stretch at least I did not find any.